tests/test_group.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+
+import greenplumpython as gp
+
+
+@pytest.fixture
+def db():
+    db = gp.database(host="localhost", dbname="gpadmin")
+    yield db
+    db.close()
+
+
+def test_group_agg(db: gp.Database):
+    rows = [(i, i % 2 == 0) for i in range(10)]
+    numbers = gp.values(rows, db=db, column_names=["val", "is_even"])
+    count = gp.aggregate("count", db=db)
+
+    results = list(numbers.group_by("is_even").aggregate(count, numbers["val"]).to_table().fetch())
```

Comment:
Why split `group_by` and `aggregate` if eventually we need to call `aggregate` function after `group_by`?

Reply:
I think separate `group by` will certainly be easier to use.